File: app/src/main/java/com/hiddenramblings/tagmo/amiibo/games/Games3DS.java
package com.hiddenramblings.tagmo.amiibo.games;

import java.util.ArrayList;

public class Games3DS {

    public final GamesManager manager;
    public final long id;
    public final ArrayList<String> games;

    public Games3DS(GamesManager manager, long id, ArrayList<String> games) {
        this.manager = manager;
        this.id = id;
        this.games = games;
    }

    public String getStringList() {
        StringBuilder usage = new StringBuilder();
        for (String game : games) {
            if (usage.length() == 0)
                usage.append("  ");
            else
                usage.append(", ");
            usage.append(game);
        }
        return usage.toString();
    }

    public boolean hasUsage(String name) {
        return null != games && !games.isEmpty() && games.contains(name);
    }
}

File: app/src/main/java/com/hiddenramblings/tagmo/amiibo/games/Games3DS.kt
package com.hiddenramblings.tagmo.amiibo.games

import com.hiddenramblings.tagmo.eightbit.io.Debug

class Games3DS(val manager: GamesManager, val id: Long, val games: ArrayList<String?>?) {
    val stringList: String
        get() {
            val usage = StringBuilder()
            games?.forEach { game ->
                if (usage.isNotEmpty()) usage.append(Debug.separator)
                usage.append(game)
            }
            return usage.toString()
        }

    fun hasUsage(name: String?): Boolean {
        return games?.contains(name) == true
    }
}